CHANGELOG.md

```diff
@@ -5,6 +5,7 @@
 * Throw a proper exception when operating on a non-existing field with the dynamic API (#3292).
 * `DynamicRealmObject.setList` should only accept `RealmList<DynamicRealmObject>` (#3280).
 * `DynamicRealmObject.getX(fieldName)` now throws a proper exception instead of a native crash when called with a field name of the wrong type (#3294).
+* Fixed a problem that `Table` and schema cache never hits against proxy classes (#3315).
```

Comment:
I would probably chuck it under "Enhancements" instead with something like "Optimized internal caching of classes" or something like that, but I don't have any strong opinions on it.